pkg/networkservice/common/capturecontext/server_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package capturecontext_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/capturecontext"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+const writeServerKey contextKeyType = "writeServerKey"
+
+type checkContextServer struct{}
+
+func (c *checkContextServer) Request(ctx context.Context, _ *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	return nil, checkContext(ctx, writeServerKey)
+}
+
+func (c *checkContextServer) Close(ctx context.Context, _ *networkservice.Connection) (*empty.Empty, error) {
+	return nil, checkContext(ctx, writeServerKey)
+}
+
+type writeServer struct{}
+
+func (w *writeServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	return next.Server(ctx).Request(capturecontext.WithCapturedContext(ctx), request)
+}
+
+func (w *writeServer) Close(ctx context.Context, connection *networkservice.Connection) (*empty.Empty, error) {
+	return next.Server(ctx).Close(capturecontext.WithCapturedContext(ctx), connection)
+}
```

Comment:
We can use adapter client to the server here to reduce code lines
```
 next.NewNetworkServiceServer(adapters.ClientToServer(&writeClient{}), capturecontext.NewServer(), adapters.ClientToServer(&checkContextClient)})
```

Reply:
Also for registry chains

Reply:
ok, fixed